### PR TITLE
refactor(button): Add ripple target as an inner element.

### PR DESCRIPTION
**Pull request by joyzhong**
BREAKING CHANGE: This changes the structure of the button element by moving the ripple from the outer <button> element to an inner `mdc-button__ripple` element.

OLD
```
<button class="mdc-button">
  <span class="mdc-button__label">Hello World</span>
</button>
```

NEW
```
<button class="mdc-button">
  <div class="mdc-button__ripple"></div>
  <span class="mdc-button__label">Hello World</span>
</button>
```

### CODE REVIEW

File: packages/mdc-button/_mixins.scss [hunk 1]
```diff
@@ -288,12 +309,17 @@
       pointer-events: none;
     }
   }
+
+  .mdc-button__label {
```
Comment (abhiomkar) [inline packages/mdc-button/_mixins.scss]: Why is this & icon positioned relative?

File: test/screenshot/spec/mdc-button/classes/baseline-button-with-icons.html [hunk 2]
```diff
@@ -44,12 +44,14 @@
       <div class="test-layout">
         <div class="test-cell test-cell--button">
           <button class="mdc-button">
+            <div class="mdc-button__ripple"></div>
```
Comment (abhiomkar) [inline test/screenshot/spec/mdc-button/classes/baseline-button-with-icons.html]: Can you also update the screenshots in spec/button/mixins folder?

File: packages/mdc-button/_mixins.scss (hunk 1 above)
Comment (joyzhong) [reply packages/mdc-button/_mixins.scss]: Removed. Originally this was added to make label/icon positioned above ripple in GM, but I'll add a separate GM mixin for that later.

File: test/screenshot/spec/mdc-button/classes/baseline-button-with-icons.html (hunk 2 above)
Comment (joyzhong) [reply test/screenshot/spec/mdc-button/classes/baseline-button-with-icons.html]: Done. Also updated screenshots for spec/dialog/mixins.

**Comment by joyzhong** [comment]
Ready for review.

**Comment by joyzhong** [comment]
BREAKING CHANGE: This changes the structure of the button element by moving the ripple from the outer <button> element to an inner `mdc-button__ripple` element.

OLD
```
<button class="mdc-button">
  <span class="mdc-button__label">Hello World</span>
</button>
```

NEW
```
<button class="mdc-button">
  <div class="mdc-button__ripple"></div>
  <span class="mdc-button__label">Hello World</span>
</button>
```

**Pull request merged**